File: basics/gcd_iterative.c
// GCD - Euclidean Algorithm (iteravive)
// Time: O(Log(n))

#include <stdio.h>

int gcd_iterative(int a, int b)
{
    while (b)
    {
        int temp = b;
        b = a % b;
        a = temp;
    }
}

// Test:
int main(void)
{
    printf("%d\n", gcd_iterative(10, 8));  // 2
    printf("%d\n", gcd_iterative(17, 13)); // 1
    return 0;
}

File: basics/gcd_iterative.py
# GCD - Euclidean Algorithm (iteravive)
# Time: O(Log(n))

def gcd_iterative(x, y):
    while y:
        x, y = y, x % y
    return x

# Test:

if __name__ == "__main__":
    print(gcd_iterative(8,3)) # 1
    print(gcd_iterative(81, 9)) # 9